add Roslyn SDK Overview document.

BillWagner opened the pull request:
This PR adds the initial Roslyn Overview pages.

Fixes dotnet/Roslyn-api-docs#16
Fixes dotnet/Roslyn-api-docs#14  

Original source is here: https://github.com/dotnet/roslyn/wiki/Roslyn%20Overview

The new TOC contains an outline for other upcoming topics, all commented out.

[Internal review link](https://review.docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/?branch=pr-en-us-3483)

/cc @jaredpar @kuhlenh

BillWagner committed:
Add the main index for roslyn conceptual

BillWagner committed:
MVC updates

BillWagner committed:
interim checkin. Switching branches

BillWagner committed:
Adding conceptual overview

BillWagner committed:
incorporate wiki content

Now, ready for an edit cycle.

BillWagner committed:
interim checkin

BillWagner committed:
ready for a test build

BillWagner commented:
WIP label added while I verify all xref links in a test build.

jaredpar commented:
CC @gafter, @VSadov and @jcouv

mairaw commented on docs/csharp/roslyn/concepts/work-with-syntax.md:
```diff
@@ -0,0 +1,102 @@
+
+# Work with Syntax
+
+The **syntax tree* is most fundamental data structure exposed by the Compiler APIs. These trees represent the lexical and syntactic structure of source code. They serve two important purposes:
```
you're missing a closing * here

mairaw commented on docs/csharp/roslyn/concepts/compiler-api-model.md:
```diff
@@ -0,0 +1,104 @@
+---
```
do you expect this area to grow a lot in size? adding levels in the url is not good for seo. also roslyn is more than just c#, no? shouldn't we have a Roslyn entry on the TOC?
Otherwise, I'd advise that we have a url like docs.microsoft.com/dotnet/roslyn and the topics directly under it (unless you expect a big expansion then having concepts is fine)

BillWagner committed:
build errors and feedback

Address initial feedback from @mairacw

Fix some of the build errors.

BillWagner commented:
closing and reopening to force a build.

BillWagner commented:
closing and reopening to force a new build

svick commented on docs/csharp/roslyn/toc.md:
```diff
@@ -0,0 +1,44 @@
+## [Understand the compiler API model](compiler-api-model.md)
+## [Work with syntax](work-with-syntax.md)
+## [Work with semantics](work-with-semantics.md)
+## [Work with a workspace](work-with-workspace.md)
```
Is documentation for Scripting APIs planned?

BillWagner replied:
Not yet. Do you want to add issues and help us design an outline?

svick commented on docs/csharp/roslyn/work-with-syntax.md:
```diff
@@ -0,0 +1,102 @@
+
+All syntax nodes are non-terminal nodes in the syntax tree, which means they always have other nodes and tokens as children. As a child of another node, each node has a parent node that can be accessed through the <xref:Microsoft.CodeAnalysis.SyntaxNode.Parent> property. Because nodes and trees are immutable, the parent of a node never changes. The root of the tree has a null parent.  
+
+Each node has a <xref:Microsoft.CodeAnalysis.SyntaxNode.ChildNodes> method, which returns a list of child nodes in sequential order based on its position in the source text. This list does not contain tokens. Each node also has methods to examine Descendants - such as <xref:Microsoft.CodeAnalysis.SyntaxNode.DescendantNodes>, <xref:Microsoft.CodeAnalysis.SyntaxNode.DescendantTokens>, or <xref:Micr ...
```
based on its position → based on their position

svick commented on docs/csharp/roslyn/work-with-syntax.md:
```diff
@@ -0,0 +1,102 @@
+
+## Spans
+
+Each node, token, or trivia knows its position within the source text and the number of characters it consists of. A text position is represented as a 32-bit integer, which is a zero-based Unicode character index. A <xref:Microsoft.CodeAnalysis.Text.TextSpan> object is the beginning position and a count of characters, both represented as integers. If <xref:Microsoft.CodeAnalysis.Text.TextSpan> ha ...
```
There's no such thing as "Unicode character". The text could talk about "UTF-16 code units", but that's probably confusing too. So what about:

```md
a zero-based `char` index
```

svick commented on docs/csharp/roslyn/work-with-syntax.md:
```diff
@@ -0,0 +1,102 @@
+
+## Kinds
+
+Each node, token, or trivia has a <xref:Microsoft.CodeAnalysis.SyntaxNode.RawKind> property, of type <xref:System.Int32?displayProperty=fullName>, that identifies the exact syntax element represented. This value can be cast to a language-specific enumeration; each language, C# or VB, has a single <xref:Microsoft.CodeAnalysis.CSharp.SyntaxKind> enumeration that lists all the possible nodes, tokens ...
```
```
each language, C# or VB, has a single <xref:Microsoft.CodeAnalysis.CSharp.SyntaxKind> enumeration
```

I find it weird that the text talks about both languages, but the link is then C#-specific. What about the following?

```
each language, C# or VB, has a single `SyntaxKind` enumeration 
(<xref:Microsoft.CodeAnalysis.CSharp.SyntaxKind?displayProperty=fullName> and 
<xref:Microsoft.CodeAnalysis.VisualBasic.SyntaxKind?displayProperty=fullName>, respectively)
```

BillWagner commented on docs/csharp/roslyn/index.md:
```diff
@@ -0,0 +1,133 @@
+use much less memory when loaded in Visual Studio than would be possible
+if you wrote your own codebase to understand the code in a project. By leveraging
+the same classes used by the compiler and Visual Studio, you can create your own
+static analysis tools. There are three main scenarios for writing analyzers and code fixes:
```
@svick those are the main use cases I've seen. If you suggest other use cases, I'll investigate and add them here.

BillWagner committed:
feedback

r+remove roslyn redirection.

BillWagner committed:
addressed feedback.

I missed these in yesterday's edits.

BillWagner commented:
closing and re-opening to force a build.

BillWagner committed:
fix xref links

BillWagner committed:
typo

BillWagner commented:
closing and reopening to force a build

BillWagner commented:
closing and reopening to start a build.

BillWagner commented:
closing and reopening to force a build.

BillWagner committed:
rename to roslyn-sdk

This works around a redirection issue, and avoids the confusion between the Roslyn SDK and the Roslyn repository.

BillWagner committed:
fix bad TOC link

BillWagner commented:
I removed the WIP label. This is ready for review.

BillWagner commented:
Adding @AnthonyDGreen to the reviewer list.

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-semantics.md:
```diff
@@ -0,0 +1,51 @@
+
+A symbol represents a distinct element declared by the source code or imported from an assembly as metadata. Every namespace, type, method, property, field, event, parameter, or local variable is represented by a symbol. 
+
+A variety of methods and properties on the <xref:Microsoft.CodeAnalysis.Compilation> type help you find symbols. For example, you can find a symbol for a declared type by its common metadata name. You can also access the entire symbol table as a tree of symbols rooted by the global namespace.
```
Do you want the xref to be the FQN instead of just type name? If so, add ?displayProperty=nameWithType

BillWagner replied:
I did mean the type name here. I thought it read more clearly. This type is abstract, and developers will use one of the derived classes (VB or C# specific).

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-semantics.md:
```diff
@@ -0,0 +1,51 @@
+
+A variety of methods and properties on the <xref:Microsoft.CodeAnalysis.Compilation> type help you find symbols. For example, you can find a symbol for a declared type by its common metadata name. You can also access the entire symbol table as a tree of symbols rooted by the global namespace.
+
+Symbols also contain additional information that the compiler determined from the source or metadata, such as other referenced symbols. Each kind of symbol is represented by a separate interface derived from <xref:Microsoft.CodeAnalysis.ISymbol>, each with its own methods and properties detailing the information the compiler has gathered. Many of these properties directly reference other symbols. ...
```
determined --> determines

rpetrusha replied:
ReturnType property of the IMethodSymbol class --> can be more succinct as \<xref:Microsoft.CodeAnalysis.IMethodSymbol.ReturnType?displayProperty=nameWithType> property

referenced --> references

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+---
+title: Use the .NET Compiler SDK syntax model 
+description: This overview provides an understanding of the type you use to understand and manipulate syntax nodes.
```
type? types?

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+# Work with Syntax
+
+The **syntax tree** is most fundamental data structure exposed by the Compiler APIs. These trees represent the lexical and syntactic structure of source code. They serve two important purposes:
```
most fundamental --> fundamental

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+Syntax trees are the primary structure used for compilation, code analysis, binding, refactoring, IDE features, and code generation. No part of the source code is understood without it first being identified and categorized into one of many well-known structural language elements. 
+
+Syntax trees have three key attributes. The first attribute is that syntax trees hold all the source information in full fidelity. This means that the syntax tree contains every piece of information found in the source text, every grammatical construct, every lexical token, and everything else in between including whitespace, comments, and preprocessor directives. For example, each literal mentio ...
```
in between including --> in between, including
no comma after malformed

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+This enables the second attribute of syntax trees. A syntax tree obtained from the parser can produce the exact text it was parsed from. From any syntax node, it is possible to get the text representation of the sub-tree rooted at that node. This means that syntax trees can be used as a way to construct and edit source text. By creating a tree you have by implication created the equivalent text,  ...
+
+The third attribute of syntax trees is that they are immutable and thread-safe.  This means that after a tree is obtained, it is a snapshot of the current state of the code, and never changes. This allows multiple users to interact with the same syntax tree at the same time in different threads without locking or duplication. Because the trees are immutable and no modifications can be made direct ...
```
the new version ---> a new version

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+## Syntax Nodes
+
+Syntax nodes are one of the primary elements of syntax trees. These nodes represent syntactic constructs such as declarations, statements, clauses, and expressions. Each category of syntax nodes is represented by a separate class derived from <xref:Microsoft.CodeAnalysis.SyntaxNode>. The set of node classes is not extensible. 
```
fully qualify SyntaxNode? (?displayProperty=nameWithType)

BillWagner replied:
This was intentional (as SyntaxNode is abstract), but I think you're right: adding the FQN.

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+Syntax nodes are one of the primary elements of syntax trees. These nodes represent syntactic constructs such as declarations, statements, clauses, and expressions. Each category of syntax nodes is represented by a separate class derived from <xref:Microsoft.CodeAnalysis.SyntaxNode>. The set of node classes is not extensible. 
+
+All syntax nodes are non-terminal nodes in the syntax tree, which means they always have other nodes and tokens as children. As a child of another node, each node has a parent node that can be accessed through the <xref:Microsoft.CodeAnalysis.SyntaxNode.Parent> property. Because nodes and trees are immutable, the parent of a node never changes. The root of the tree has a null parent.  
```
\<xref:Microsoft.CodeAnalysis.SyntaxNode.Parent?displayProperty=nameWithType>

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+All syntax nodes are non-terminal nodes in the syntax tree, which means they always have other nodes and tokens as children. As a child of another node, each node has a parent node that can be accessed through the <xref:Microsoft.CodeAnalysis.SyntaxNode.Parent> property. Because nodes and trees are immutable, the parent of a node never changes. The root of the tree has a null parent.  
+
+Each node has a <xref:Microsoft.CodeAnalysis.SyntaxNode.ChildNodes> method, which returns a list of child nodes in sequential order based on their position in the source text. This list does not contain tokens. Each node also has methods to examine Descendants - such as <xref:Microsoft.CodeAnalysis.SyntaxNode.DescendantNodes%2A>, <xref:Microsoft.CodeAnalysis.SyntaxNode.DescendantTokens%2A>, or <x ...
```
include type name with ChildNodes
Descendants - such --> descendants, such
the descendants methods should include type names
- that --> , that

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+For efficiency purposes, the <xref:Microsoft.CodeAnalysis.SyntaxToken> type is a CLR value type. Therefore, unlike syntax nodes, there is only one structure for all kinds of tokens with a mix of properties that have meaning depending on the kind of token that is being represented.
+
+For example, an integer literal token represents a numeric value. In addition to the raw source text the token spans, the literal token has a Value property that tells you the exact decoded integer value. This property is typed as Object because it may be one of many primitive types.
```
Value property: \<xref:Microsoft.CodeAnalysis.SyntaxToken.Value>
Object: \<xref:System.Object>?

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+The <xref:Microsoft.CodeAnalysis.SyntaxToken.ValueText> property tells you the same information as the <xref:Microsoft.CodeAnalysis.SyntaxToken.Value> property; however this property is always typed as <xref:System.String>. An identifier in C# source text may include Unicode escape characters, yet the syntax of the escape sequence itself is not considered part of the identifier name. So although  ...
+
+For example, if the source text contains identifier written as `\u03C0`, then
```
No need for a paragraph break here.

contains identifier --> contains an identifier

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+You can access trivia by inspecting a token’s <xref:Microsoft.CodeAnalysis.SyntaxToken.LeadingTrivia> or <xref:Microsoft.CodeAnalysis.SyntaxToken.TrailingTrivia> collections. When source text is parsed, sequences of trivia are associated with tokens. In general, a token owns any trivia after it on the same line up to the next token. Any trivia after that line is associated with the following toke ...
+
+Unlike syntax nodes and tokens, syntax trivia do not have parents. Yet, because they are part of the tree and each is associated with a single token, you may access the token it is associated with using the <xref:Microsoft.CodeAnalysis.SyntaxTrivia.Token> property.
```
use displayProperty=nameWithType with Token property

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+Unlike syntax nodes and tokens, syntax trivia do not have parents. Yet, because they are part of the tree and each is associated with a single token, you may access the token it is associated with using the <xref:Microsoft.CodeAnalysis.SyntaxTrivia.Token> property.
+
+Like syntax tokens, trivia are value types. The single <xref:Microsoft.CodeAnalysis.SyntaxTrivia> type is used to describe all kinds of trivia.
```
I think this is better at the end of the first paragraph.

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+Because trivia are not part of the normal language syntax and can appear anywhere between any two tokens, they are not included in the syntax tree as a child of a node. Yet, because they are important when implementing a feature like refactoring and to maintain full fidelity with the source text, they do exist as part of the syntax tree.
+
+You can access trivia by inspecting a token’s <xref:Microsoft.CodeAnalysis.SyntaxToken.LeadingTrivia> or <xref:Microsoft.CodeAnalysis.SyntaxToken.TrailingTrivia> collections. When source text is parsed, sequences of trivia are associated with tokens. In general, a token owns any trivia after it on the same line up to the next token. Any trivia after that line is associated with the following toke ...
```
the two xrefs should be displayProperty=nameWithType

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+Each node, token, or trivia has a <xref:Microsoft.CodeAnalysis.SyntaxNode.RawKind> property, of type <xref:System.Int32?displayProperty=fullName>, that identifies the exact syntax element represented. This value can be cast to a language-specific enumeration; each language, C# or VB, has a single `SyntaxKind` enumeration  (<xref:Microsoft.CodeAnalysis.CSharp.SyntaxKind?displayProperty=fullName> a ...
+
+The <xref:Microsoft.CodeAnalysis.SyntaxToken.RawKind> property allows for easy disambiguation of syntax node types that share the same node class. For tokens and trivia, this property is the only way to distinguish one type of element from another. 
```
RawKind should be ?displayProperty=nameWithType

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+Even when the source text contains syntax errors, a full syntax tree that is round-trippable to the source is exposed. When the parser encounters code that does not conform to the defined syntax of the language, it uses one of two techniques to create a syntax tree.
+
+First, if the parser expects a particular kind of token, but does not find it, it may insert a missing token into the syntax tree in the location that the token was expected. A missing token represents the actual token that was expected, but it has an empty span, and its <xref:Microsoft.CodeAnalysis.SyntaxNode.IsMissing> property returns `true`.
```
no comma before but
IsMissing should be ?displayProperty=nameWithType

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-syntax.md:
```diff
@@ -0,0 +1,106 @@
+
+First, if the parser expects a particular kind of token, but does not find it, it may insert a missing token into the syntax tree in the location that the token was expected. A missing token represents the actual token that was expected, but it has an empty span, and its <xref:Microsoft.CodeAnalysis.SyntaxNode.IsMissing> property returns `true`.
+
+Second, the parser may skip tokens until it finds one where it can continue parsing. In this case, the skipped tokens that were skipped are attached as a trivia node with the kind <xref:Microsoft.CodeAnalysis.CSharp.SyntaxKind.SkippedTokensTrivia>.
```
skipped tokens that were skipped --> skipped token

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-workspace.md:
```diff
@@ -0,0 +1,43 @@
+---
+title: Work with the .NET Compiler SDK workspace model
+description: This overview provides an understanding of the type you use to understand and manipulate the workspace and projects for your code.
```
duplication of understand/understanding

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-workspace.md:
```diff
@@ -0,0 +1,43 @@
+
+# Work with a Workspace
+
+The **Workspaces** layer is the starting point for doing code analysis and refactoring over entire solutions. Within this layer, the Workspace API assists you in organizing all the information about the projects in a solution into single object model, offering you direct access to compiler layer object models like source text, syntax trees, semantic models and compilations without needing to pars ...
```
into single --> into a single
comma after configure options

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-workspace.md:
```diff
@@ -0,0 +1,43 @@
+
+A workspace is an active representation of your solution as a collection of projects, each with a collection of documents. A workspace is typically tied to a host environment that is constantly changing as a user types or manipulates properties. 
+
+The <xref:Microsoft.CodeAnalysis.Workspace> provides access to the current model of the solution. When a change in the host environment occurs, the workspace fires corresponding events, and the <xref:Microsoft.CodeAnalysis.Workspace.CurrentSolution> property is updated. For example, when the user types in a text editor corresponding to one of the source documents, the workspace uses an event to s ...
```
CurrentSolution can be nameWithType
by making --> making

rpetrusha commented on docs/csharp/roslyn-sdk/work-with-workspace.md:
```diff
@@ -0,0 +1,43 @@
+
+## Summary
+
+Roslyn exposes a set of Compiler APIs and Workspaces APIs that provides rich information about your source code and that has full fidelity with the C# and Visual Basic languages.  The .NET Compiler Platform SDK dramatically lowers the barrier to entry for creating code focused tools and applications. It creates many opportunities for innovation in areas such as meta-programming, code generation a ...
```
code focused --> code-focused

rpetrusha reviewed (changes requested):
I've left a large number of comments, @BillWagner, but this is really good content.

BillWagner commented on docs/csharp/roslyn-sdk/compiler-api-model.md:
```diff
@@ -0,0 +1,104 @@
+---
+title: .NET compiler SDK concepts and object model
```
I went through all the files and standardized this way:
. The NuGet package and product name is ".NET Compiler Platform". When I refer to that SDK, that's the phrase I use.
. Other instances of "compiler" mean the C# or VB compiler, and are lower case.

BillWagner commented on docs/csharp/roslyn-sdk/index.md:
```diff
@@ -0,0 +1,143 @@
+ms.custom: mvc
+---
+
+# The .NET compiler platform SDK
```
Upper case here.

BillWagner commented on docs/csharp/roslyn-sdk/index.md:
```diff
@@ -0,0 +1,143 @@
+
+## .NET Compiler SDK Concepts
+
+The .NET Compiler SDK provides access to the deep understanding that
```
I re-wrote the first few sentences.

BillWagner commented on docs/csharp/roslyn-sdk/index.md:
```diff
@@ -0,0 +1,143 @@
+increase our productivity. We rely on code analysis tools to improve our
+code quality and code generators to aid in application construction. As
+these tools get smarter, they need access to more and more of the deep
+code knowledge that only compilers possess. This is the core mission of
```
re-worked to match the introductory paragraph.

BillWagner commented on docs/csharp/roslyn-sdk/index.md:
```diff
@@ -0,0 +1,143 @@
+generation and transformation, interactive use of the C# and VB languages,
+and embedding of C# and VB in domain specific languages.
+
+The .NET compiler platform SDK enables you to build ***analyzers*** and 
```
I standardized on ".NET Compiler Platform SDK" throughout this PR.

BillWagner commented on docs/csharp/roslyn-sdk/index.md:
```diff
@@ -0,0 +1,143 @@
+standard guideline validation.
+
+The .NET Compiler platform SDK provides a single set of APIs that enable
+you to examine and understand a C# or Visual Basic codebase. Because you
```
codebase

BillWagner commented on docs/csharp/roslyn-sdk/index.md:
```diff
@@ -0,0 +1,143 @@
+have created analyzers that enforce those recommended patterns. As we learn
+more, there is always room for new ideas.
+
+These analyzers can be uploaded to the Visual Studio Gallery and downloaded
```
And the branding changed...

BillWagner commented on docs/csharp/roslyn-sdk/work-with-semantics.md:
```diff
@@ -0,0 +1,51 @@
+
+For example, many types, fields, methods, and local variables with the same name may be spread throughout the source. Although each of these is uniquely different, determining which one an identifier actually refers to often requires a deep understanding of the language rules. 
+
+There are program elements represented in source code, and programs can also refer to previously compiled libraries, packaged in assembly files. Although no source code is available for assemblies and therefore no syntax nodes or trees, programs can still refer to elements inside them.
```
I think I reworked this section correctly. Let me know if I understood your comments correctly.

BillWagner committed:
respond to feedback

mairaw commented on docs/csharp/roslyn-sdk/index.md:
```diff
@@ -0,0 +1,146 @@
+
+# The .NET Compiler Platform SDK
+
+## .NET Compiler Platform SDK Concepts
```
usually you have an introduction before jumping to a H2. Maybe move this to after the first paragraph?

also sentence case

mairaw commented on docs/csharp/roslyn-sdk/index.md:
```diff
@@ -0,0 +1,146 @@
+journey. As these become more widely used, the community adopts these
+practices.
+
+## Next Steps
```
nit: sentence case for headings

BillWagner replied:
updated throughout the PR

mairaw commented on docs/csharp/roslyn-sdk/compiler-api-model.md:
```diff
@@ -0,0 +1,106 @@
+
+![compiler pipeline](media/compiler-pipeline.png)
+
+Each phase of this pipeline is a separate component. First the
```
nit: missing comma after first

mairaw commented on docs/csharp/roslyn-sdk/compiler-api-model.md:
```diff
@@ -0,0 +1,106 @@
+in the code to symbols. Finally, the emit phase, emits an assembly with
+all the information built up by the compiler.
+
+![compiler pipeline api](media/compiler-pipeline-api.png)
```
alt text should be a bit more descriptive. If you couldn't see the image, how would you describe that image?

BillWagner replied:
updated all image elements in this PR

mairaw commented on docs/csharp/roslyn-sdk/compiler-api-model.md:
```diff
@@ -0,0 +1,106 @@
+
+In addition, the Workspaces layer surfaces a set of APIs used
+when implementing code analysis and refactoring tools that function within
+a host environment like the Visual Studio IDE. Excamples include the Find All References,
```
typo: Excamples

mairaw reviewed (changes requested):
LGTM. Left a few comments and also have some general comments/questions:
- Headings should use sentence case
- Breadcrumb file should also be updated
- If this works with VB too, is under the csharp folder the right place to be?

BillWagner commented:
@mairaw said:

> - If this works with VB too, is under the csharp folder the right place to be?

I discussed this with the product team, and they felt it was the right place. The VB development community has had much smaller interest.

BillWagner committed:
respond to feedback.

mairaw commented:
Thanks for clarifying @BillWagner! I thought it was sort of a dumb question initially. 😄

mairaw reviewed (approved):
LGTM. The headings that had .NET Compiler Platform SDK, that's ok to use the proper case. I was referring to the other words in the headings.

BillWagner committed:
upper case ".NET Compiler Platform SDK

BillWagner commented:
I'll merge this.  @mairaw reviewed that I had made all the changes requested by @rpetrusha 

Enjoy the holidays @rpetrusha
